### PR TITLE
docs: update docs around column aliasing

### PR DESCRIPTION
**Pull request by big-andy-coates**
### Description 

System generated column aliases are now guaranteed to be unique.

Also raised https://github.com/confluentinc/ksql/issues/5428 while in the file.

### Testing done 

doc change.

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: docs/developer-guide/ksqldb-reference/select-push-query.md [hunk 1]
```diff
@@ -29,17 +29,23 @@ Description
...
-Note that the WINDOW clause can only be used if the `from_item` is a
-stream.
+the console, or returned to the client. To stop a push query started in the CLI press Ctrl+C.
 
-Push queries enable you to query a materialized view with a subscription to
-the results. Push queries emit refinements to materialized views, which enable
+Execute a push query via the CLI or by sending an HTTP request to the ksqlDB REST API, and
+the API sends back a chunked response of indefinite length.
+
+Push queries enable you to subscribe to changes, which enable
```
Comment (big-andy-coates) [inline docs/developer-guide/ksqldb-reference/select-push-query.md:37]: I've changed this from:

> Push queries enable you to query a materialized view

As that would suggest you can only perform push queries on MVs.... which will suggest tables to most users, while ksqlDB supports push queries on streams too.

File: docs/developer-guide/ksqldb-reference/select-push-query.md [hunk 2]
```diff
@@ -29,17 +29,23 @@ Description
...
 reacting to new information in real-time. They’re a good fit for asynchronous
 application flows. For request/response flows, see [Pull Queries](select-pull-query.md).
 
-Execute a push query by sending an HTTP request to the ksqlDB REST API, and
-the API sends back a chunked response of indefinite length.
+Push queries can use all available SQL features, which can be useful when prototyping a
+persistent query or when running ad-hoc queries from the CLI. However, unlike persistent queries,
```
Comment (JimGalasyn) [inline docs/developer-guide/ksqldb-reference/select-push-query.md:42]:
```suggestion
persistent query or when running ad-hoc queries from the CLI. But unlike persistent queries,

```

File: docs/developer-guide/ksqldb-reference/select-push-query.md [hunk 3]
```diff
@@ -29,17 +29,23 @@ Description
...
+push queries are not shared. If multiple clients submit the same push query ksqlDB will compute
```
Comment (JimGalasyn) [inline docs/developer-guide/ksqldb-reference/select-push-query.md:43]:
```suggestion
push queries are not shared. If multiple clients submit the same push query, ksqlDB computes

```

File: docs/developer-guide/ksqldb-reference/select-push-query.md [hunk 4]
```diff
@@ -29,17 +29,23 @@ Description
...
+independent results for each client.
+
+!!! tip
+  If using push queries from an application, move all the heavy lifting into a persistent query
```
Comment (JimGalasyn) [inline docs/developer-guide/ksqldb-reference/select-push-query.md:47]:
```tip
  If you're using push queries from an application, move all the heavy lifting into a persistent query

```

File: docs/developer-guide/ksqldb-reference/select-push-query.md [hunk 5]
```diff
@@ -107,6 +113,9 @@ SET 'auto.offset.reset' = 'earliest';
 
 #### WINDOW
 
+!!! note
+  WINDOW clause can only be used if the `from_item` is a stream.
```
Comment (JimGalasyn) [inline docs/developer-guide/ksqldb-reference/select-push-query.md:117]:
```suggestion
  You can use the WINDOW clause only if the `from_item` is a stream.

```

**Comment by big-andy-coates** [comment]
> LGTM, with a couple of suggestions. Looks like this should get back-ported to the live branches, correct?

The fix that ensures system generated column names are unique is only in master, so ... not everything...

**Pull request merged**